src/wasm-emscripten.h

```diff
@@ -18,27 +18,47 @@
 #define wasm_wasm_emscripten_h
 
 #include "wasm.h"
+#include "wasm-builder.h"
 
 namespace wasm {
 
 class LinkerObject;
 
-namespace emscripten {
+class EmscriptenGlueLinker {
```

Comment:
Why is this called "linker"? There should probably be some more comments here. This header is the most important part of the PR because this class will persist after asm2wasm and s2wasm are gone.

Reply:
Because it generates emscripten glue, and used to operate on the wasm linker...

Suggestions welcome please

Reply:
EmscriptenGlueGenerator?